build/Build.cs

```diff
@@ -696,6 +701,27 @@ string GetNugetVersion()
                   ?? throw new InvalidOperationException("Unable to retrieve valid Nuget Version");
         }
 
+        void CreateHashFileForProject(string directory)
+        {
+            byte[] GetHashOfFileContent(string f)
+            {
+                using var md5 = MD5.Create();
+                using var stream = File.OpenRead(f); 
+                return md5.ComputeHash(stream);
+            }
+
+            var allFiles = Directory.GetFiles(directory, "*.*", SearchOption.AllDirectories);
+
+            var concatFileHashes = allFiles.SelectMany(GetHashOfFileContent);
```

Comment:
This only has the hash of files, not their filenames & path... I think you probably want something like:
- filepath1/hash1
- filepath2/hash2

then sort and concat that result.

Reply:
yeah - could, figured this would be safe, without going too far down the rabbit hole.

Reply:
:shrug: its a cheap change, and better represents the application "shape" - see if I can do it.